src/axom/core/Array.hpp

```diff
@@ -1164,6 +1170,8 @@ Array<T, DIM, SPACE>::Array(Array&& other) noexcept
   m_capacity = other.m_capacity;
   m_resize_ratio = other.m_resize_ratio;
   m_allocator_id = other.m_allocator_id;
+  m_executeOnGPU = axom::detail::getAllocatorSpace(m_allocator_id) ==
```

Comment:
Suggestion: This is a one-liner, but might is somewhat tricky to get right, and perhaps subject to change, should it be a function here or in axom/core/memory_management?

Reply:
Hi Kenny, good idea. I'm doing that now. I was running into trouble on CI anyway with MemorySpace::Device only existing when UMPIRE is enabled.

Reply:
As for a reproducer, I could not get it to happen with the test I wrote (so I junked it). The fixes do clear up my problems on CUDA though on my MIR branch.

Reply:
From CI:
```
2024-10-01T19:41:44.0228174Z /home/axom/axom/src/axom/core/../../axom/core/Array.hpp:293:28: error: no member named 'Device' in 'axom::MemorySpace'
2024-10-01T19:41:44.0244447Z         axom::MemorySpace::Device;
2024-10-01T19:41:44.0245077Z         ~~~~~~~~~~~~~~~~~~~^
```
https://dev.azure.com/healy22/d4445b8f-6e95-49cd-9e5b-8c8ca1ccad31/_apis/build/builds/10807/logs/58

More reason to move this to a function. `Device` is only defined when Umpire is available.
See: https://github.com/LLNL/axom/blob/70b360815ebed6a25e6ae369bc5efeaa58cacdbc/src/axom/core/memory_management.hpp#L37-L47

+1 for the team for having CI configs w/o Umpire!

Reply:
> As for a reproducer, I could not get it to happen with the test I wrote (so I junked it). The fixes do clear up my problems on CUDA though on my MIR branch.

Probably still good to have test cases that exercise the `true` and `false` branches